connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskMockitoTest.java

```diff
@@ -0,0 +1,763 @@
+                            : Collections.emptyMap()
+            );
+        };
+    }
+
+    private void expectConversionAndTransformation(final String topicPrefix, final Headers headers) {
+        when(keyConverter.toConnectData(TOPIC, headers, RAW_KEY)).thenReturn(new SchemaAndValue(KEY_SCHEMA, KEY));
+        when(valueConverter.toConnectData(TOPIC, headers, RAW_VALUE)).thenReturn(new SchemaAndValue(VALUE_SCHEMA, VALUE));
+
+        for (Header header : headers) {
+            when(headerConverter.toConnectHeader(TOPIC, header.key(), header.value())).thenReturn(new SchemaAndValue(VALUE_SCHEMA, new String(header.value())));
+        }
+
+        expectTransformation(topicPrefix);
+    }
+
+    private void expectTransformation(final String topicPrefix) {
+        when(transformationChain.apply(any(SinkRecord.class))).thenAnswer((Answer<SinkRecord>)
+                invocation -> {
+                    SinkRecord origRecord = invocation.getArgument(0);
+                    return topicPrefix != null && !topicPrefix.isEmpty()
+                            ? origRecord.newRecord(
+                            topicPrefix + origRecord.topic(),
+                            origRecord.kafkaPartition(),
+                            origRecord.keySchema(),
+                            origRecord.key(),
+                            origRecord.valueSchema(),
+                            origRecord.value(),
+                            origRecord.timestamp(),
+                            origRecord.headers()
+                    ) : origRecord;
+                });
+    }
+
+    private void expectTaskGetTopic() {
+        when(statusBackingStore.getTopic(anyString(), anyString())).thenAnswer((Answer<TopicStatus>) invocation -> {
+            String connector = invocation.getArgument(0, String.class);
+            String topic = invocation.getArgument(1, String.class);
+            return new TopicStatus(topic, new ConnectorTaskId(connector, 0), Time.SYSTEM.milliseconds());
+        });
+    }
+
+    private void assertSinkMetricValue(String name, double expected) {
+        MetricGroup sinkTaskGroup = workerTask.sinkTaskMetricsGroup().metricGroup();
+        double measured = metrics.currentMetricValueAsDouble(sinkTaskGroup, name);
+        assertEquals(expected, measured, 0.001d);
+    }
+
+    private void assertTaskMetricValue(String name, double expected) {
+        MetricGroup taskGroup = workerTask.taskMetricsGroup().metricGroup();
+        double measured = metrics.currentMetricValueAsDouble(taskGroup, name);
+        assertEquals(expected, measured, 0.001d);
+    }
+
+    private void assertTaskMetricValue(String name, String expected) {
+        MetricGroup taskGroup = workerTask.taskMetricsGroup().metricGroup();
+        String measured = metrics.currentMetricValueAsString(taskGroup, name);
+        assertEquals(expected, measured);
+    }
+
+    private void printMetrics() {
```

Comment:
cleanup: this method and children are (correctly) unused in the new class. The metric assertions should provide debug-ability by printing the actual measured values already.

Reply:
removed